### PR TITLE
(#1513) Replace `console` with `client-logger` and `node-logger` packages

### PR DESCRIPTION
**Pull request by dangreenisrael**
Issue: #1513

## What I did

Added a logger package containing a logger to be use in the browser and another to be used in node.

Note: This is for work in progress. Any feedback is more than welcome.

## How to test

Is this testable with jest or storyshots?
Jest

Does this need a new example in the kitchen sink apps?
No

Does this need an update to the documentation?
Yes, it will need to be documented for add-on developers

If your answer is yes to any of these, please make sure to include it in your PR.

### Node
screen shot 2017-11-23 at 11 24 28 am

### Browser

screen shot 2017-11-24 at 10 24 27 pm

### CODE REVIEW

**Comment by dangreenisrael** [comment]
There seems to be some sort of issue with the build step of the integration tests that happening on circle but not local for me 😕 

It ends with
```
Storybook started on => http://localhost:9009/

Step was canceled
```

**Comment by Hypnosphi** [comment]
> This is for `app/react` only

Why?

File: app/react/src/client/preview/render.js [hunk 1]
```diff
@@ -3,14 +3,13 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
 import { stripIndents } from 'common-tags';
+import logger from 'npmlog';
```
Comment (Hypnosphi) [inline app/react/src/client/preview/render.js]: Does it work on client?

**Comment by dangreenisrael** [comment]
I just wanted to make sure that this was what the team wanted, before making the change throughout the ecosystem.

**Comment by dangreenisrael** [comment]
@Hypnosphi @ndelangen Do you guys prefer to import `npmlog` directly into files, or have a `storybook-client-logger` and `storybook-node-logger` that we keep in a central location and use throughout the mono-repo?

File: app/react/src/client/preview/render.js (hunk 1 above)
Comment (dangreenisrael) [reply app/react/src/client/preview/render.js]: (facepalm) It was late when I was working on this. Thanks @Hypnosphi

**Comment by ndelangen** [comment]
> @Hypnosphi @ndelangen Do you guys prefer to import `npmlog` directly into files, or have a `storybook-client-logger` and `storybook-node-logger` that we keep in a central location and use throughout the mono-repo?

Sounds like a good thing indeed, if we want to change logger, we can do it in a central location 👍

**Comment by ndelangen** [comment]
@dangreenisrael I usually take some screenshots of the changes just before opening up the PR, so I can post them.

At least for me, it helps visualise and remember what the PR is about, what it's doing.

👍

**Review by Hypnosphi — APPROVED** [review]

**Comment by Hypnosphi** [comment]
@dangreenisrael feel free to merge if you think it's ready

**Comment by dangreenisrael** [comment]
@Hypnosphi Do you think I should roll the new logger out across the ecosystem or do each platform in its own PR?

**Comment by Hypnosphi** [comment]
As you wish. I think there’s a sense in adding this to all the packages that log anything

**Comment by dangreenisrael** [comment]
@Hypnosphi I think I would prefer to do each platform in its own, smaller PR. I think that keeping PRs small makes things more manageable, and makes it easier to track down bugs if they are introduced.

**Comment by Hypnosphi** [comment]
Will it be a separate PR for each addon as well?

**Comment by dangreenisrael** [comment]
Not all of the addons use console.log. I will probably do one PR for the rest of the platforms, and one PR for the rest of the addons. Also, if I run out of time doing this, I want it to be as easy as possible for someone else to takeover.

**Pull request merged**